terraform/20-app/main.tf

```diff
@@ -28,3 +35,53 @@ terraform {
     key            = "app/state.tfstate"
   }
 }
+
+module "cognito" {
+  source = "../modules/cognito"
+
+  user_pool_name    = "app-${terraform.workspace}-user-pool"
+  client_name       = "app-${terraform.workspace}-client"
+  user_pool_domain  = "app-${terraform.workspace}-auth"
+
+  callback_urls = ["https://${terraform.workspace}.dev.ukhsa-dashboard.data.gov.uk/callback"]
+  logout_urls   = ["https://${terraform.workspace}.dev.ukhsa-dashboard.data.gov.uk/logout"]
+}
+
+resource "aws_security_group" "app_security_group" {
+  name        = "app-security-group"
+  description = "Security group for the application"
+  vpc_id = module.vpc.vpc_id
+
+  ingress {
+    description      = "Allow traffic from app to RDS"
+    from_port        = 5432
+    to_port          = 5432
+    protocol         = "tcp"
+    security_groups  = [module.aurora_db_app.security_group_id]
+    self             = false
+  }
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+}
+
+output "rds_connection_info" {
```

Comment:
It feels a little odd to be outputting things which belong to another module?
As in here module A is outputted things from module B. But they're mutually exclusive?